release/models/system/openconfig-gnsi-authz.yang

```diff
@@ -0,0 +1,178 @@
+    }
+
+    revision 2022-01-17 {
+        description
+            "Initial revision.";
+        reference "0.1.0";
+    }
+
+    typedef version {
+        type string;
+        description
+            "The version ID of the gRPC authorization policy as provided by
+             the gRPC Authorization Policy Manager when the policy was pushed.
+             This leaf persists through a reboot.";
+    }
+
+    typedef created-on {
+        type oc-types:timeticks64;
+        description
+            "The creation time of the gRPC authorization policy as reported by
+             the gRPC Authorization Policy manager when the policy was pushed
+             to the device. This value is reported as nanoseconds since epoch
+             (January 1st, 1970 00:00:00 GMT). This leaf persists through
+             a reboot.";
+    }
+
+    // gRPC server authorization policy related definitions.
+    grouping counters {
+        description
+          "A collection of counters that were collected by the gNSI.authz
+           module while evaluating access to a RPC.";
+
+        leaf access-rejects {
+            type oc-yang:counter64;
+            description
+              "The total number of times the gNSI.authz module denied access
+               to a RPC.";
+        }
+        leaf last-access-reject {
+            type oc-types:timeticks64;
+            description
+              "A timestamp of the last time the gNSI.authz denied access to
+               a RPC.";
+        }
+        leaf access-accepts {
+            type oc-yang:counter64;
+            description
+              "The total number of times the gNSI.authz module allowed access
+               to a RPC.";
+        }
+        leaf last-access-accept {
+            type oc-types:timeticks64;
+            description
+              "A timestamp of the last time the gNSI.authz allowed access to
+               a RPC.";
+        }
+    }
+
+    grouping grpc-server-user-authz-policy-success-failure-counters {
+        description
+          "A collection of counters collected by the gNSI.authz module.";
+
+        container rpcs {
+            description
+              "A collection of counters collected by the gNSI.authz module
+               for each RPC separately.";
+
+            list rpc {
+                description
+                  "A collection of counters collected by the gNSI.authz module
+                   for a RPC identified by the `name`.";
+
+                key name;
+                leaf name {
+                    type leafref {
+                        path "../state/name";
+                    }
+                    description
+                      "The name of the RPC the counters were collected for.";
+                }
+                container state {
+                    leaf name {
+                        type string;
+                        description
+                            "The name of the RPC the counters were collected
```

Comment:
Are you suggesting that the server might be using a non-RPC mechanism here? authz was designed for RPCs: https://github.com/openconfig/gnsi/tree/main/authz#gnsiauthz

Reply:
No, just stating consistently specifying the encoding of the RPC name (sorry my comment on the proto fix was back towards acctz specifically)

https://github.com/openconfig/gnsi/blob/main/acctz/acctz.proto#L240-L244

To be encoded as such:

```
  Header: :path: /gnmi.gNMI/Subscribe
      Name Length: 5
      Name: :path
      Value Length: 20
      Value: /gnmi.gNMI/Subscribe
      :path: /gnmi.gNMI/Subscribe
      [Unescaped: /gnmi.gNMI/Subscribe]
      Representation: Literal Header Field with Incremental Indexing - Indexed Name
      Index: 5
```

Otherwise, we may see a variation of how "RPCs" are encoded here... I'll take care of the proto PR

Reply:
ref: https://github.com/openconfig/gnsi/pull/146

Reply:
Interesting, so the one that I've been using for NOSImageProfile's OCRPCs comes from https://pkg.go.dev/google.golang.org/protobuf/reflect/protoreflect#FullName, documented here: https://github.com/openconfig/featureprofiles/blob/91e38dc16ad1a8e3b91c322d3fdefe7778808a69/proto/ocrpcs.proto#L34-L39

I think we should have some consistency on this.

Seeing gRPC CLI uses the dot format I'd propose to use the same: https://github.com/grpc/grpc/blob/master/doc/command_line_tool.md#list-methods

Reply:
Yea that's more or less my meta point is to converge to some consistency and inspect what is in use when - I naturally went towards how things are encoded in the HTTP/2 header expected here

https://github.com/grpc/grpc/blob/master/doc/PROTOCOL-HTTP2.md

And if we dissect the visibility components have in a pipeline, you'll have some that are unaware of the protobuf encoding doing plain HTTP2/gRPC call routing where there is accounting/logging and will be based on the :path header format.

Speaking on behalf of our current implementation, AAA, proxy routing, logging, etc... is all centered around the HTTP/2 :path format - anything else is likely to cause format conversions across the board

Reply:
This sounds good to me: it will create some pain if people were to use a tool that exclusively used `.`, but it is clearer, and at least there is a one-way conversion. I'll modify the description as such and add an example for others to comment on.